docs/e2e/requirements.md

```diff
@@ -0,0 +1,38 @@
+#### Real RP environment requirements
+
+The following environment variables are required for running the e2e tests against the real RP
+
+| Name | Notes |
+|--- | --- |
+| `AZURE_CLIENT_ID` | authentication |
+| `AZURE_CLIENT_SECRET` | authentication |
+| `AZURE_TENANT_ID` | authentication |
+| `AZURE_SUBSCRIPTION_ID` | existing cluster |
+| `AZURE_REGION` | existing cluster location |
+| `RESOURCEGROUP` | existing cluster group |
+
+#### Fake RP environment requirements
+
+The following environment variables are required for running the e2e tests against the fake RP
+
+| Name | Notes |
+|--- | --- |
+| `AZURE_CLIENT_ID` | authentication |
+| `AZURE_CLIENT_SECRET` | authentication |
+| `AZURE_TENANT_ID` | authentication |
+| `AZURE_SUBSCRIPTION_ID` | existing cluster |
+| `AZURE_REGION` | existing cluster location |
+| `RESOURCEGROUP` | existing cluster group |
+
+The following are the files required for running the e2e tests against the fake RP
+
+| Name | Notes |
+|--- | --- |
+| `_data/containerservice.yaml` | allows the test to determine if it is an update or not |
```

Comment:
Hrm, if this is needed only for determining whether we are on an update vs create, I would try to think how to get rid of it.

Reply:
at a minimum it is required to discriminate against an update vs create. However, it is also needed for certain admin-oriented, update-based e2e tests such as key rotation.